### PR TITLE
sphobjinv-textconv

### LINKED ISSUE #295: act as textconv so can git diff inventory
**Is your feature request related to a problem? Please describe.**
git can be informed on which files are binary. And then instructed how to convert the binary files into text files, so ```git diff``` can work. 

git passes in the .inv file name cuz the ``git diff`` is occurring on the .inv, not on the .txt file.

```sphobjinv convert plain``` requires a hyphen to be passed in to have the output sent to stdout. Instead would like a option flag. So then sphobjinv can act as a textconv

docs/.gitattributes

```
[diff "inv"]
  textconv = sphobjinv --stdout convert plain 
  binary = true
```

**Describe the solution you'd like**
Provide a --stdout option flag as another means to specify the outfile hyphen.

**Describe alternatives you've considered**
Alternative approach, which i'm using now, is to mark the inventory files as binary and not be able to use ```git diff```

docs/.gitattributes

```
*.inv binary
```

**Additional context**

#159 is a similar feature request, but for stdin
The difference is infile is a required positional argument whereas outfile is optional. So a --stdout flag would not change sphobjinv command line interface

[perform diffs on binary files](https://git-scm.com/docs/gitattributes#_performing_text_diffs_of_binary_files)

the fallback, [indicate files are binary](https://git-scm.com/docs/gitattributes#_using_macro_attributes)

### PR DESCRIPTION
**Pull request by msftcangoblowm**
**Is the PR a fix or a feature?**
feature

**Describe the changes in the PR**
Adds entrypoint sphobjinv-textconv

**Does this PR close any issues?**
Closes #295

**Does the PR change/update the following, if relevant?**

- azure-pipelines adjust coverage fail under reduced 100% --> 90%. Purposefully in a separate commit, for later easy adjustment via `git rebase`

- No CHANGELOG entry. No adjustment to version

- `pyproject.toml` reformatted by pre-commit

- Unit tests. No integration test. Some fixtures added to `conftest.py`

- Minimal adjustments to `doc/conf.py` to pass linkcheck.

- `# pragma: no cover` used within the entrypoint, not for non soi-textconv modules. In non soi-textconv modules, using `# pragma: no cover` would return coverage to ~99%. 

- [*] Documentation
- [*] Tests
- [  ] CHANGELOG

### CODE REVIEW

**Comment by bskinn** [comment]
Sorry for the delay here, needed to release #299 to make the package play nicer with Debian's build process, and then the holidays hit.

First thing -- your fork's `main` still isn't current with upstream, which might be the cause of the merge conflict:

image

It's possible you didn't push your updated `main` to your fork?  If `git diff origin/main..upstream/main` shows any changes, then something's off.

File: conftest.py [hunk 1]
```diff
@@ -82,6 +82,51 @@ def res_dec(res_path, misc_info):
     return res_path / (misc_info.FNames.RES.value + misc_info.Extensions.DEC.value)
 
 
+@pytest.fixture(scope="session")
+def res_cmp_plus_one_line(res_path, misc_info):
```
Comment (bskinn) [inline conftest.py:86]: Is this fixture needed any more, since this PR will only be adding the `sphobjinv-textconv` endpoint and not actually doing anything with git diffs?

File: conftest.py [hunk 2]
```diff
@@ -264,6 +303,68 @@ def func(arglist, *, expect=0):  # , suffix=None):
     return func
 
 
+@pytest.fixture()  # Must be function scope since uses monkeypatch
+def run_cmdline_textconv(monkeypatch):
```
Comment (bskinn) [inline conftest.py:307]: Instead of duplicating this fixture with a different hard-coded command, I would rather generalize the existing `run_cmdline_test()` by having it not assume what the command is, and have all of the points-of-use of the fixture just pass in the command along with the other arguments.

(It's possible that this isn't feasible, I haven't looked closely. Interested in your thoughts.)

File: conftest.py [hunk 3]
```diff
@@ -264,6 +303,68 @@ def func(arglist, *, expect=0):  # , suffix=None):
...
+
+        # Test that execution completed w/indicated exit code
+        assert retcode == expect, runargs
+
+    return func
+
+
+@pytest.fixture()  # Must be function scope since uses monkeypatch
+def run_cmdline_no_checks(monkeypatch):
+    """Return function to perform command line. So as to debug issues no tests."""
```
Comment (bskinn) [inline conftest.py:341]: Is this still needed?

File: conftest.py [hunk 4]
```diff
@@ -152,19 +197,13 @@ def scratch_path(tmp_path, res_path, misc_info, is_win, unix2dos):
     # With the conversion of resources/objects_attrs.txt to Unix EOLs in order to
     # provide for a Unix-testable sdist, on Windows systems this resource needs
     # to be converted to DOS EOLs for consistency.
-    if is_win:
+    if is_win:  # pragma: no cover
```
Comment (bskinn) [inline conftest.py:200]: I'm confused as to why these lines started showing up in any of the coverage testing. My pipelines don't show either `setup.py` or `conftest.py`, in either the [GH Action][gha] for testing the main code or the [Azure Pipeline][ap] for checking execution of all the test code.

These pragmas shouldn't be necessary.

[ap]: https://dev.azure.com/brianskinn/sphobjinv/_build/results?buildId=564&view=logs&jobId=a97395d9-d56f-57fb-fb5c-9044ddefd9a7&j=a97395d9-d56f-57fb-fb5c-9044ddefd9a7&t=89437840-2e4d-5005-e67a-78711bfacfe0
[gha]: https://github.com/bskinn/sphobjinv/actions/runs/12458286225/job/34773701964#step:7:42

File: tests/resource/objects_attrs_plus_one_entry.inv
Comment (bskinn) [inline tests/resource/objects_attrs_plus_one_entry.inv:1]: Is this file needed if we're not directly testing `git diff` with textconv?

File: tests/resource/objects_attrs_plus_one_entry.txt
Comment (bskinn) [inline tests/resource/objects_attrs_plus_one_entry.txt:1]: Is this file needed if we're not directly testing `git diff` with textconv?

**Review by bskinn — COMMENTED** [review]
Incomplete review; couple of high-level comments/questions for you.

**Comment by msftcangoblowm** [comment]
Dear Brian,

Was wondering why the commits/PR was taken out of chronological order. As long as there is a good reason.

> First thing -- your fork's main still isn't current with upstream

Before a merge, since #299 occurred, so will have to rebase [msftcangoblowm:textconv](https://github.com/msftcangoblowm/sphobjinv/tree/textconv)Might be able to just rebase [msftcangoblowm:textconv](https://github.com/msftcangoblowm/sphobjinv/tree/textconv) and continue ignoring [msftcangoblowm:](https://github.com/msftcangoblowm/sphobjinv/tree/textconv)main
[msftcangoblowm/sphobjinv:textconv](https://github.com/msftcangoblowm/sphobjinv/tree/textconv)Have never done a serious rebase
> It's possible you didn't push your updated main to your fork

That's most definitely the case and might not matter at all. What's important is your main. And keeping up to date with your main. That's what a rebase will solve.

Taken directly from the #301 PR merge page

`[msftcangoblowm](https://github.com/msftcangoblowm) wants to merge 4 commits into [bskinn:main](https://github.com/bskinn/sphobjinv/tree/main) from [msftcangoblowm:textconv](https://github.com/msftcangoblowm/sphobjinv/tree/textconv)`

Working in a branch is normal.

After the merge, can resync both [msftcangoblowm:](https://github.com/msftcangoblowm/sphobjinv/tree/textconv)main and [msftcangoblowm:textconv](https://github.com/msftcangoblowm/sphobjinv/tree/textconv)

Will do the rebase and message you (on mastodon) once that's done.

Dave
